### ISSUE TITLE
Sync Immediately / Synchronization Difficulties

### ISSUE THREAD

**Issue by The-Maxx** [Tampermonkey/tampermonkey#862]
(Please fill out the issue template with your details)

## Expected Behavior

1. Modify and save a script on one computer.
2. Wait for messages "sync running, change exported, sync finished".

- View the script list on a second computer and see that the script has been updated without any further user interaction..
OR
- Press the "Check for userscript updates" option and see a message "All scripts have been updated."
AND
- Any scripts removed from the second computer that exist on the first computer are restored.

## Actual Behavior

1. Modify and save a script on one computer.
2. Wait for messages "sync running, change exported, sync finished".

- View the script list on a second computer and see that the script has NOT been updated without any further user interaction.
AND
- Press the "Check for userscript updates" options and see two messages "Check for userscript updates..." and "No updates found, sry!". (Although after that and refreshing the page, the list of scripts shows that the script has actually been updated.)
AND
- Any scripts removed from the second computer that exist on the first computer are NOT restored.

## Specifications

First Computer:
- Chromium: Google Chrome Version 79.0.3945.130 (Official Build) (64-bit)
- TM: v4.9
- OS: Windows 10 Pro Version 1903 OS Build 18362.418 64-bit

Second Computer:
- Chromium: Google Chrome Version 79.0.3945.123 (Official Build) {32-bit)
- TM: v4.9
- OS: Google Chrome OS Version 79.0.3945.123 (Official Build) (32-bit)

Script Sync Method: Google Drive

## Script

N/A

----------

So basically I would like sync across devices to happen immediately (if possible) when the extension is opened and loaded.
OR
Have the messages that are displayed be more accurate after clicking the "Check for userscript updates" option.
AND
After clicking the "Check for userscript updates" option, reload any scripts that have been removed but still actually exist elsewhere.

Thanks a lot and keep up the great work!  :):):)

P.S. Could you make it so that when one word is selected in the Tampermonkey text editor, all the other instances of the same word are also highlighted?

**Comment by The-Maxx** [comment]
Any update on this?  I got it to sync at the time of reporting the issue, but now I can't seem to get it to sync again even after following some of that steps that seemed to work before.

**Comment by derjanb** [comment]
> Press the "Check for userscript updates" options and see two messages "Check for userscript updates..." and "No updates found, sry!". (Although after that and refreshing the page, the list of scripts shows that the script has actually been updated.)

This really updates userscripts from their origins. As a "side effect" a sync is also triggered first in order to not update scripts (which updates their modification date) that are deleted at the cloud and now would be restored due to this update.

> Any scripts removed from the second computer that exist on the first computer are NOT restored.

When exactly was the script removed. I'd expect the change of deletion to by synced to the second computer if the script was deleted when sync was enabled.

> View the script list on a second computer and see that the script has NOT been updated without any further user interaction.

In order to not overwhelm given cloud storage request limits sync is not done instantly but periodically.

> After clicking the "Check for userscript updates" option, reload any scripts that have been removed but still actually exist elsewhere.

I don't get this. Sorry.

BTW: there was a bug that sometimes the script source was not updated correctly which is fixed at [Tampermonkey BETA](https://chrome.google.com/webstore/detail/tampermonkey-beta/gcalenpjmijncebpfijmoaglllgpjagf).

**Comment by The-Maxx** [comment]
Hi, thanks for responding.  Things in general are all a bit confusing sometimes, so it can be difficult to explain.  I think I understand what you mean, but sometimes it seems to work and sometimes not, but I tried out many variations to get it to work to end up seeing that the sync was turned off on the source device, so after turning that back on it ended up syncing properly to the target device.  Maybe if it happens again I can try to be more accurate and post a more specific message.  Thanks again.  BTW - Any thoughts about the idea of adding the text highlighting feature I recommended?  (Could you make it so that when one word is selected in the Tampermonkey text editor, all the other instances of the same word are also highlighted?)  thx :)

**Comment by derjanb** [comment]
> Could you make it so that when one word is selected in the Tampermonkey text editor, all the other instances of the same word are also highlighted?

#893
